wrangler-service/src/main/java/co/cask/wrangler/service/directive/DirectivesService.java

```diff
@@ -594,38 +594,34 @@ public void execute(HttpServiceRequest request, HttpServiceResponder responder,
         }
       });
 
-      JsonArray values = new JsonArray();
-      JsonArray headers = new JsonArray();
-      JsonObject types = new JsonObject();
-      Set<String> header = new HashSet<>();
+      List<Map<String, Object>> values = new ArrayList<>(rows.size());
+      Map<String, String> types = new HashMap<>();
+      Set<String> headers = new LinkedHashSet<>();
```

Comment:
why is it changed to LinkedHashSet?

Reply:
The column headers need to preserve order, so using a LinkedHashSet. The previous implementation using a Set in combination with a JsonArray was essentially implementing the relevant LinkedHashSet behavior.